winmake.ps1

```diff
@@ -1,4 +1,5 @@
-$ErrorActionPreference = 'Stop'
+
+. ./contrib/cirrus/win-lib.ps1
```

Comment:
Not an expert on Windows but now becoming a hobbyist, but is there anything we're setting/enabling/disabling in win-lib.ps1 that we don't necessarily want running locally, such as the env vars? As in if someone ran `winmake` on their local system to run tests, would it mess with their current powershell setup?

Reply:
You and me both...it's a steep (and very frustrating) hill :frowning_face: 

You make a good point, I don't think so?  I believe most of those env. vars. are all scoped as one would expect (see below).  Ahh, but there's some clearing of golang env. vars. that's only relevant for CI.  I'll wrap those in a conditional.

I re-read the docs on this just now.  It seems some variables can be user and/or machine scoped, [but a special SetEnvironmentVariable syntax](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.3#saving-environment-variables-with-setenvironmentvariable) is required ,that win-lib doesn't use.

The "preferences" env. vars. are definitely scoped to only the script process and its children.  [Thank $DEITY that's actually documented.](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.3#powershell-environment-variables)

Reply:
I think @ashley-cui has a good point about the local build use case and the CI case having some differences. As an example the win-lib library does reasonable CI things like disables color. So it might make sense to break these into two scripts one that is a ci-lib and the other that is common, perhaps ci-lib includes the common one. I see you already updated it to guard with a CI flag so maybe largely dealt with, but longer term perhaps that should be thought about.

(BTW sorry to reopen just wanted to ensure my comment was visible)

Reply:
> perhaps ci-lib includes the common one.

That's a good idea actually, thanks for sharing.